src/lib/components/drs/DrsMain.js

```diff
@@ -0,0 +1,292 @@
+    parents: _.pick(formProps, ['drs_object_parents', 'addParent', 'setParentId', 'setParentName', 'setParentValid', 'setParentInvalid', 'unsetParentValidity', 'removeParent']),
+    fileAccessObjects: _.pick(formProps, ['file_access_objects', 'addFileAccessObject', 'setFileAccessObjectPath', 'removeFileAccessObject']),
+    awsS3AccessObjects: _.pick(formProps, ['aws_s3_access_objects', 'addAwsS3AccessObject', 'setAwsS3AccessObjectRegion', 'setAwsS3AccessObjectBucket', 'setAwsS3AccessObjectKey', 'removeAwsS3AccessObject']),
+    submit: {activeDrsObject: activeDrsObject, setSuccessMessage: setSuccessMessage, ..._.pick(formProps, ['retrieveDrsObjectsList'])}
+  }
+
+  /* ##################################################
+   * EFFECTS
+   * ################################################## */
+
+  // add the history listener, modifies the activeDrsObject based on the current
+  // URL path
+  let history = useHistory();
+  useEffect(() => {
+    console.log('running effect a');
+    history.listen((location, action) => {
+      let re = new RegExp('/drs/(.+)');
+      let match = location.pathname.match(re);
+      if (match) {
+        match[1] === 'new' ? activeDrsObjectFunctions.reset() : activeDrsObjectFunctions.retrieve(match[1]);
+      }
+    })
+  }, [])
+
+  // initialize the drsObjectList upon first load
+  useEffect(() => {
+    console.log('running effect b');
+    retrieveDrsObjectsList();
+  }, []);
+
+  // update the allowed checksum types list every time any checksum type
+  // changes
+  useEffect(() => {
+    let updatedDisplayChecksumTypes = emptyDisplayChecksumTypes();
+    activeDrsObject.checksums.forEach((checksum, index) => {
+      updatedDisplayChecksumTypes[checksum.type] = index;
+    })
+    setDisplayChecksumTypes(updatedDisplayChecksumTypes);
+  }, [activeDrsObject]);
+
+  /* ##################################################
+   * RENDER
+   * ################################################## */
+
+  return (
+    <div>
+      <Snackbar
+        anchorOrigin={{ vertical: 'top', horizontal: 'center'}}
+        open={successMessage}
+        onClose={() => setSuccessMessage(null)}
+      >
+        <Alert onClose={() => setSuccessMessage(null)} severity="success" variant="filled">
+          {successMessage ? successMessage : null }
+        </Alert>
+    </Snackbar>
+
+      <Switch>
+        <Route exact path='/drs'>
+          <DrsIndex 
+            drsObjectsList={drsObjectsList} 
+            setError={setError}
+          />
+        </Route>
+        <Route exact path='/drs/new'>
+          <DrsObjectForm
+            title={"Create New DrsObject"}
+            groupedFormProps={groupedFormProps}
+            formViewType={FormViewType.NEW}
+          />
+        </Route>
+        <Route exact path='/drs/:objectId'>
+          <DrsObjectForm
+            title={`View DrsObject: ${activeDrsObject.id}`}
+            groupedFormProps={groupedFormProps}
+            formViewType={FormViewType.SHOW}
```

Comment:
Upon viewing the SHOW page for a given DRS Object, the scroll is not automatically restored to the top of the page. For example, if the user selects an object that is at the bottom of the Index page, the SHOW page will render at the bottom of the SHOW page as well.